db/db.go

```diff
@@ -0,0 +1,57 @@
+package db
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+var db *gorm.DB
+
+func OpenDatabase(dsn string) error {
+	it, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	if err != nil {
+		return err
+	}
+	db = it
+
+	err = db.AutoMigrate(
+		&User{},
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type User struct {
+	ID         int `gorm:"primaryKey"`
+	Disc_id    int
+	First_name string
+	Last_name  string
+	Dob        time.Time
+}
```

Comment:
```suggestion
type User struct {
	Id         int `gorm:"primaryKey"`
	DiscordId    int
	FirstName string
	LastName  string
	Dob        time.Time
}
```

if gorm complains about fields (or they come back empty) you need to use a struct field tag to override the column name (which is explained in the gorm docs [here](https://gorm.io/docs/conventions.html#Column-Name))